lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,449 @@
+
+                * name: The name of the team.
+
+                * is_active: True if the team is currently active. If false, the team is considered "soft deleted" and
+                  will not be included by default in results.
+
+                * course_id: The identifier for the course this team belongs to.
+
+                * topic_id: Optionally specifies which topic the team is associated with.
+
+                * date_created: Date and time when the team was created.
+
+                * description: A description of the team.
+
+                * country: Optionally specifies which country the team is associated with.
+
+                * language: Optionally specifies which language the team is associated with.
+
+                * membership: A list of the users that are members of the team. See membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in, a 403 error is returned.
+
+            If the specified course_id is not valid or the user attempts to
+            use an unsupported query parameter, a 400 error is returned.
+
+            If the response does not exist, a 404 error is returned. For
+            example, the course_id may not reference a real course or the page
+            number may be beyond the last page.
+
+        **Response Values for POST**
+
+            Any logged in user who has verified their email address can create
+            a team. The format mirrors that of a GET for an individual team,
+            but does not include the id, is_active, date_created, or membership
+            fields. id is automatically computed based on name.
+
+            If the user is not logged in or has not verified their email, a
+            403 error is returned.
+
+            If the course_id is not valid or extra fields are included in the
+            request, a 400 error is returned.
+
+            If the specified course does not exist, a 404 error is returned.
+    """
+
+    # SessionAuthentication must come first to return a 403 for unauthenticated users
+    authentication_classes = (SessionAuthentication, OAuth2Authentication)
+    permission_classes = (permissions.IsAuthenticated, IsActiveOrReadOnly)
+
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = CourseTeamSerializer
+
+    def get_serializer_context(self):
+        """
+        Adds expand information from query parameters to the serializer context to support expandable fields.
+        """
+        result = super(TeamsListView, self).get_serializer_context()
+        result['expand'] = [x for x in self.request.QUERY_PARAMS.get('expand', '').split(',') if x]
```

Comment:
If it's only membership that uses "expand", then it is fine to leave it out. Please add a note in the JIRA ticket though so we don't forget to do it-- https://openedx.atlassian.net/browse/TNL-1907

In fact, it would be good to clarify in TNL-1907 what is left to be done, since parts are already implemented.